kwarchart/src/main/java/com/kwarchart/android/chart/BarChart.kt

```diff
@@ -1,18 +1,277 @@
+import android.graphics.PorterDuff
+
+import android.graphics.PorterDuffColorFilter
+import androidx.compose.ui.graphics.*
+import androidx.compose.ui.graphics.painter.ColorPainter
+
+import com.kwarchart.android.*
+import com.kwarchart.android.*
+
+
+private var mHGap = 0f
+private var mVGap = 0f
+private var mMaxVal = 0f
+private var mMaxLen = 0
 
 /**
  *
  */
 @Composable
-fun BarChart() {
+fun <T> BarChart(
+        modifier: Modifier = Modifier,
+        data: List<BarSeries<T>>,
+        xAxisName: String? = null,
+        yAxisName: String? = null,
+        axesColor: Color = Color.Gray,
+        showAxes: Boolean = true,
+        gridsColor: Color = Color.Gray,
+        showGrid: Boolean = true,
+        legendPos: LegendPosition? = null
+) {
+    val keys = mutableListOf<T>()
+
+    data.forEach {
+        if (mMaxLen < it.data.size) {
+            mMaxLen = it.data.size
+        }
+
+        it.data.forEach { chartData ->
+            if (mMaxVal < chartData.value) {
+                mMaxVal = chartData.value
+            }
+
+            if (!keys.contains(chartData.key)) {
+                keys.add(chartData.key)
+            }
+        }
+    }
+
+    Chart(
+            modifier = modifier,
+            xAxisName = xAxisName,
+            yAxisName = yAxisName,
+            legend = legendPos?.let { _ ->
+                val tmpLegend = Legend(legendPos)
+                data.forEach {
+                    tmpLegend.legends.add(it.legend)
+                    tmpLegend.colors.add(it.color)
+                }
+                tmpLegend
+            }
+    ) {
+        Canvas(
+                modifier = modifier
+                        .weight(5f)
+                        .padding(
+                                start = (mMaxVal.toString().length * 8).dp,
+                                top = 10.dp,
+                                end = 10.dp,
+                                bottom = AXIS_VALUES_FONT_SIZE.dp
+                        )
+        ) {
+            mHGap = size.width / mMaxLen
+            mVGap = size.height / mMaxLen
+
+            if (showGrid) {
+                drawGrids(mMaxLen, gridsColor)
+            }
+            if (showAxes) {
+                drawAxes(axesColor, keys)
+            }
+            data.forEach {
+                drawData(it)
+            }
+        }
+    }
+}
+
+
+/**
+ * Plot line chart data.
+ *
+ * @param barSeries Data to be plotted in this chart.
+ */
+private fun <T> DrawScope.drawData(barSeries: BarSeries<T>) {
+    barSeries.data.forEachIndexed { i, chartData ->
+        val point = getDataPoint(i, chartData, size.height)
+        drawRect(
+                topLeft = Offset(point.x - 24, point.y),
+                size = Size(width = 48f, height = size.height - point.y),
+                color = barSeries.color,
+                style = Stroke(barSeries.width.toFloat()),
+//              brush = Brush.linearGradient(colors = listOf(barSeries.areaColor!!))
```

Comment:
remove if not used

Reply:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.